core/src/prover.rs

```diff
@@ -60,11 +60,46 @@ impl Prover for NativeProver {
         Ok(Proof {
             proof: None,
             quote: None,
-            kzg_proof: None,
         })
     }
 
     async fn cancel(_proof_key: ProofKey, _read: Box<&mut dyn IdStore>) -> ProverResult<()> {
         Ok(())
     }
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_native_prover() {
```

Comment:
Need to keep this? Native prover already being tested in the different block proving tests.

Reply:
I wanna keep it cuz this pull deserialize GuestInput from the the json path in `data/`, which shows you the exact block & network it's processing independent of the RPC. Like if we change execution logic that's incompatible with ethererum this will immediately fail

Reply:
If the input format is changed, people will have to regenerate an input json for this test to make it work again, so will require extra work to maintain tests which is why I'm always hesitant of doing this (and this code doesn't go through the standard raiko input caching mechanism, so require some manual setup). Maybe this can be an ignored test, so people can use it if they want but is not used by CI (and CI will still fail with RPC based block tests)?